crates/optimism/payload/src/builder.rs

```diff
@@ -620,7 +625,15 @@ where
         let base_fee = builder.evm_mut().block().basefee;
 
         while let Some(tx) = best_txs.next(()) {
+            // We skip invalid cross chain tx, they would be removed on the next block update in the
+            // maintenance job
+            if let Some(interop) = tx.interop() {
+                if !interop.is_valid(self.config.attributes.timestamp()) {
+                    continue
```

Comment:
we need to mark this one as invalid, so that the iter clears the sender's txs that we know won't be valid:

best_txs.mark_invalid(tx.signer(), tx.nonce());

we also should only do this after all the less expensive checks after the 4844 sanity check on L646